src/fabric/src/fabric_doc_open_revs.erl

```diff
@@ -458,6 +464,21 @@ check_finish_quorum_newer(_) ->
         meck:history(fabric)
     ).
 
+check_finish_quorum_replicator(_) ->
+    % We count a descendant of a revision for quorum so
+    % foo1 should count for foo2 which means we're finished.
+    % Since we use a replicator shard read_repair doesn't trigger
+    W1 = #shard{node = 'node1'},
+    W2 = #shard{node = 'node2'},
+    S0 = state0(all, false),
+    S1 = S0#state{dbname = <<"foo/_replicator">>},
+    {ok, S2} = handle_message({ok, [foo1(), bar1()]}, W1, S1),
+    Expect = {stop, [bar1(), foo2()]},
+    ok = meck:reset(fabric),
+    ?assertEqual(Expect, handle_message({ok, [foo2(), bar1()]}, W2, S2)),
+    timer:sleep(100),
```

Comment:
Could we also use `meck:wait` here if we add some logging above like I suggested?

Reply:
It's a fairly short sleep and I think it works well for the "ensure async function _wasn't_ called" pattern.